homeassistant/components/nextcloud/entity.py

```diff
@@ -1,31 +1,65 @@
 """Base entity for the Nextcloud integration."""
-
+from typing import Any
+from urllib.parse import urlparse
 
 from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.helpers.device_registry import DeviceEntryType
 from homeassistant.helpers.entity import DeviceInfo
 from homeassistant.helpers.update_coordinator import CoordinatorEntity
 
 from .const import DOMAIN
 from .coordinator import NextcloudDataUpdateCoordinator
 
+DIAGNOSTICS = ["apcu", "cache", "interned_strings", "jit", "opcache", "sma"]
+ICONS = {
+    "activeUsers": "mdi:account-multiple",
+    "database": "mdi:database",
+    "freespace": "mdi:harddisk",
+    "mem_free": "mdi:memory",
+    "mem_total": "mdi:memory",
+    "php": "mdi:language-php",
+    "swap_free": "mdi:memory",
+    "swap_total": "mdi:memory",
+    "update": "mdi:update",
+}
+
 
 class NextcloudEntity(CoordinatorEntity[NextcloudDataUpdateCoordinator]):
     """Base Nextcloud entity."""
 
     _attr_has_entity_name = True
-    _attr_icon = "mdi:cloud"
 
     def __init__(
-        self, coordinator: NextcloudDataUpdateCoordinator, item: str, entry: ConfigEntry
+        self,
+        coordinator: NextcloudDataUpdateCoordinator,
+        item: str,
+        entry: ConfigEntry,
+        attrs: dict[str, dict[str, Any]] | None = None,
     ) -> None:
         """Initialize the Nextcloud sensor."""
         super().__init__(coordinator)
+        domain = urlparse(coordinator.url).netloc
         self.item = item
         self._attr_name = item
-        self._attr_unique_id = f"{coordinator.url}#{item}"
+        self._attr_unique_id = f"{domain}#{item}"
```

Comment:
You cannot change the unique_id without making this a breaking change.
If you insist on making this change then please open a dedicated PR just for the `unique_id` change.